src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs

```diff
@@ -40,6 +40,8 @@ private sealed class State
             public MsQuicConnection.State ConnectionState = null!; // set in ctor.
             public string TraceId = null!; // set in ctor.
 
+            public uint StartStatus = MsQuicStatusCodes.Success;
```

Comment:
Is it intentional that it will be "success" even before the event arrives?

Reply:
Yes, we check it only in shutdown complete to unblock pending operations with proper exception. There's no point in distinguishing between start not called yet and  called and succeeded.
I also assume this will get reworked with #67302.